rust/src/dns/dns.rs

```diff
@@ -421,7 +437,12 @@ pub(crate) fn dns_parse_request(input: &[u8]) -> Result<DNSTransaction, DNSParse
             if request.invalid_authorities {
                 tx.set_event(DNSEvent::InvalidAuthorities);
             }
-            tx.request = Some(request);
+
+            if variant.is_mdns() && request.header.flags & 0x8000 != 0 {
+                tx.response = Some(request);
+            } else {
+                tx.request = Some(request);
+            }
```

Comment:
I wonder if flipping the direction during probe, to fake responses as `to_client` could get rid of this, and the variant types. Will try. Worried about side affects tho, as there is no real `to_client` traffic in mDNS.